Assets/_Application/Layers/LayerTypes/RootLayer.cs

```diff
@@ -123,6 +124,21 @@ public void UpdateLayerTreeOrder(int index)
             }
         }
         
+        public ReferencedLayerData GetFirstLayerByLayerMask(LayerMask mask)
+        {
+            foreach (LayerData data in ChildrenLayers)
```

Comment:
`return ChildrenLayers.TypeOf<ReferencedLayerData>().FirstOrDefault(refData => refData.Reference.gameObject.layer == mask);` is voldoende hier :)